parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -1062,34 +1063,34 @@ prettyParseError s = \case
   go (Parser.UnknownDataConstructor    tok _referents) = unknownConstructor "data" tok
   go (Parser.UnknownId               tok referents references) = Pr.lines
     [ if missing then
-        "I couldn't resolve the reference " <> style ErrorSite (HQ.toString (L.payload tok)) <> "."
+        "I couldn't resolve the reference " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> "."
       else
-        "The reference " <> style ErrorSite (HQ.toString (L.payload tok)) <> " was ambiguous."
+        "The reference " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> " was ambiguous."
     , ""
-    , tokenAsErrorSite s $ HQ.toString <$> tok
+    , tokenAsErrorSite s $ HQ.toString . fmap Name.fromVar <$> tok
     , if missing then "Make sure it's spelled correctly."
       else "Try hash-qualifying the term you meant to reference."
     ]
     where missing = Set.null referents && Set.null references
   go (Parser.UnknownTerm               tok referents) = Pr.lines
     [ if Set.null referents then
-        "I couldn't find a term for " <> style ErrorSite (HQ.toString (L.payload tok)) <> "."
+        "I couldn't find a term for " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> "."
       else
-        "The term reference " <> style ErrorSite (HQ.toString (L.payload tok)) <> " was ambiguous."
+        "The term reference " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> " was ambiguous."
     , ""
-    , tokenAsErrorSite s $ HQ.toString <$> tok
+    , tokenAsErrorSite s $ HQ.toString . fmap Name.fromVar <$> tok
     , if missing then "Make sure it's spelled correctly."
       else "Try hash-qualifying the term you meant to reference."
     ]
     where
     missing = Set.null referents
   go (Parser.UnknownType               tok referents) = Pr.lines
     [ if Set.null referents then
-        "I couldn't find a type for " <> style ErrorSite (HQ.toString (L.payload tok)) <> "."
+        "I couldn't find a type for " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> "."
       else
-        "The type reference " <> style ErrorSite (HQ.toString (L.payload tok)) <> " was ambiguous."
+        "The type reference " <> style ErrorSite (HQ.toString (Name.fromVar <$> L.payload tok)) <> " was ambiguous."
     , ""
-    , tokenAsErrorSite s $ HQ.toString <$> tok
+    , tokenAsErrorSite s $ HQ.toString . fmap Name.fromVar <$> tok
     , if missing then "Make sure it's spelled correctly."
       else "Try hash-qualifying the type you meant to reference."
     ]
```

Comment:
I understand that this code is converting some `v`s into `Name`s (as they were `Name`s here before the PR), but I already forget what that means.  

What's the relationship between `Name`, `v`, custom syntax, and what should get displayed here?

If `Name` is just "thing that represents what's stored in the `Names`pace", then we maybe don't need to convert to `Name` here (although it's fine as an incremental change, regardless), since we're not dealing with `Names`.  Or if the pluggable syntax tells us how to render `Name`, then maybe that's a fine type to use here.

But is there a difference between the syntax's rendered `Name` and the original `v`?

Reply:
Good questions. My guess is many of these don't have to be converted to `Name`, they are just reusing printing/convenience functions that already operate on `Name` like `HQ.toString`.